controllers/machine_controller.go

```diff
@@ -261,7 +261,8 @@ func (r *MachineReconciler) reconcileDelete(ctx context.Context, cluster *cluste
 	logger := r.Log.WithValues("machine", m.Name, "namespace", m.Namespace)
 	logger = logger.WithValues("cluster", cluster.Name)
 
-	if err := r.isDeleteNodeAllowed(ctx, m); err != nil {
+	isDeleteNodeAllowed, err := r.isDeleteNodeAllowed(ctx, m)
```

Comment:
It feels odd to use the value of `isDeleteNodeAllowed` when it is only false if we return an error. It might be better to use a boolean that is set and manipulated outside of the call to `isDeleteAllowed`.

Reply:
make sense, done. 
@detiber can you help understand the reasoning behind deleting the infra even for cases where `isDeleteNodeAllowed` is false?

Reply:
@enxebre If we block the infrastructure deletion we are essentially blocking deletion of the Cluster. The original idea behind isDeleteNodeAllowed was to avoid failures when we knew that node deletion was likely to fail, not to prevent the deletion of the infrastructure backing the Machine.